scripts/common/proxy.sh

```diff
@@ -1,25 +1,78 @@
 
+function proxy_bootstrap() {
+    if [ "$AIRGAP" = "1" ]; then
+        return
+    fi
+    if curl --silent --connect-timeout 4 --fail https://api.replicated.com/market/v1/echo/ip > /dev/null ; then
+        return
+    fi
+    # Need to peek at the yaml spec to find if a proxy is needed to download the util binaries
+    if [ -n "$INSTALLER_SPEC_FILE" ]; then
+        local overrideProxy=$(grep "proxyAddress:" "$INSTALLER_SPEC_FILE" | grep -o "http[^'\" ]*")
+        if [ -n "$overrideProxy" ]; then
+            export https_proxy="$overrideProxy"
+            echo "Bootstrapped proxy address from installer spec file: $https_proxy"
+            return
+        fi
+    fi
+    local proxy=$(echo "$INSTALLER_YAML" | grep "proxyAddress:" | grep -o "http[^'\" ]*")
+    if [ -n "$proxy" ]; then
+        export https_proxy="$proxy"
+        echo "Bootstrapped proxy address from installer yaml: $https_proxy"
+        return
+    fi
+
+    bail "Failed to make outbound https request and no proxy is configured."
+}
+
 function configure_proxy() {
-    exportProxy
-    # kubeadm requires this in the environment to reach the K8s API server
-    export no_proxy="$NO_PROXY_ADDRESSES"
+    if [ "$NO_PROXY" = "1" ]; then
```

Comment:
This is after the binaries have property parsed the yaml spec and updated the relevant environment variables.